pdns/opensslsigners.cc

```diff
@@ -1233,39 +1408,118 @@ std::string OpenSSLECDSADNSCryptoKeyEngine::sign(const std::string& message) con
 
 bool OpenSSLECDSADNSCryptoKeyEngine::verify(const std::string& message, const std::string& signature) const
 {
-  if (signature.length() != (d_len * 2)) {
+  if (signature.length() != (static_cast<unsigned long>(d_len) * 2)) {
     throw runtime_error(getName() + " invalid signature size " + std::to_string(signature.length()));
   }
 
-  string l_hash = this->hash(message);
+  // NOLINTNEXTLINE(*-cast): Using OpenSSL C APIs.
+  auto* signatureCStr = const_cast<unsigned char*>(reinterpret_cast<const unsigned char*>(signature.c_str()));
+  auto rComponent = BigNum(BN_bin2bn(signatureCStr, d_len, nullptr), BN_clear_free);
```

Comment:
Given that this is a signature (not secret) we could use `BN_free` instead to avoid zeroing the memory.